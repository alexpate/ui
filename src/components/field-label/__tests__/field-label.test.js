/* @flow */
/* eslint-env jest */
import expect from 'expect';
import {shallow} from 'enzyme';
import {textMatch} from 'utils/test/enzyme';
import {testComponent} from 'utils/test/react';

import FieldLabel from '../field-label';
import Text from '../../text';
import Icon from '../../icon';

describe('components/field-label', () => {
  describe('FieldLabel', () => {
    const create = testComponent(FieldLabel, () => ({
      htmlFor: 'user-email',
      children: 'Your email',
    }));

    it('should render shallow component ok', () => {
      const {element} = create();
      const result = shallow(element);
      expect(result).toExist('should render OK');
    });

    it('should render with a required asterisk', () => {
      const {element} = create({
        required: true,
      });
      const result = shallow(element);
      const label = result.find(Text);
      expect(textMatch(label, 'Your email *')).toBe(true);
    });

    it('should render with a locked icon', () => {
      const {element} = create({
        locked: true,
      });
      const result = shallow(element);
      expect(result.find(Icon).length).toBe(1);
    });
  });
});
