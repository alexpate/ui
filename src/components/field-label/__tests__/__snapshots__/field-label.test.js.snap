// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldLabel should render shallow component ok 1`] = `
<label
  className="css-16ef8bd0 css-1i78l7h"
  color="navy600"
  htmlFor="user-email"
  onClick={undefined}
  size="small"
  style={Object {}}
  title={undefined}
>
  Your email
</label>
`;
